### PR TITLE
Nearby Flight app

### PR DESCRIPTION
**Pull request by kevinelliott**
The following adds the ability to show a nearby flight (sometimes overhead) of your location using data from a private Airframes API (https://airframes.io) made specifically for the Tidbyt. I am the founder of Airframes, so I am authorized to release this.

nearby_flight

### CODE REVIEW

**Comment by kevinelliott** [comment]
I have read the CLA Document and I hereby sign the CLA

File: apps/nearbyflight/nearby_flight.star
Comment (matslina) [inline apps/nearbyflight/nearby_flight.star:31]: We'll need these coordinates truncated before submitting to the API. Even though I'm sure the API provider is most trustworthy, we try to guarantee that exact user locations aren't shared unless it's absolutely necessary. For this use case, I think truncating to 3 decimals would be good enough? That would provide resolution of around 100m.

File: apps/nearbyflight/nearby_flight.star
Comment (matslina) [inline apps/nearbyflight/nearby_flight.star:27]: Unlike in pixlet, `secret.decrypt()` failure will actually result in a hard error when running in our backend environment. Since `ENCRYPTED_API_KEY` is blank, this will always fail.

File: apps/nearbyflight/nearby_flight.star
Comment (matslina) [inline apps/nearbyflight/nearby_flight.star:142]: Probably want to add some caching around this call too? Just to keep request rates down. Your call, since you're running the API. =)

File: apps/nearbyflight/nearby_flight.star
Comment (matslina) [inline apps/nearbyflight/nearby_flight.star:186]: Note that these field description are currently not visible to users in the Tidbyt iOS/Android app. You'll need to add the information to the app description in the manifest file if you want people to see it.

File: apps/nearbyflight/nearby_flight.star
Comment (matslina) [inline apps/nearbyflight/nearby_flight.star:33]: You may want to `return []` here instead. That will send a "blank screen" to the user, which results in the app not being displayed in the rotation.

File: apps/nearbyflight/nearby_flight.star
Comment (matslina) [inline apps/nearbyflight/nearby_flight.star:26]: Would be great if you could add an illustrative default graphic for the case when no config has been provided. That will allow us to display a preview of the app so that users get a sense of whether or not this app (and airframes.io) is something for them.

**Review by matslina — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks very cool!

A few comments inline that we should probably address before merging.

**Comment by kevinelliott** [comment]
Thank you @matslina for the comments and review. I will look at the feedback and begin to address.

File: apps/nearbyflight/nearby_flight.star
Comment (kevinelliott) [reply apps/nearbyflight/nearby_flight.star:31]: The locations are already obfuscated.

File: apps/nearbyflight/nearby_flight.star
Comment (kevinelliott) [reply apps/nearbyflight/nearby_flight.star:142]: Cached over here, but yeah, I will consider!

File: apps/nearbyflight/nearby_flight.star
Comment (kevinelliott) [reply apps/nearbyflight/nearby_flight.star:186]: Definitely, makes sense.

File: apps/nearbyflight/nearby_flight.star
Comment (kevinelliott) [reply apps/nearbyflight/nearby_flight.star:33]: Good idea, thanks for the thoughts here. Definitely think a "No nearby flight" or similar message is best.

File: apps/nearbyflight/nearby_flight.star
Comment (kevinelliott) [reply apps/nearbyflight/nearby_flight.star:26]: Oh interesting. Let me think about this.

File: apps/nearbyflight/nearby_flight.star
Comment (matslina) [reply apps/nearbyflight/nearby_flight.star:31]: Could you elaborate a bit on this? I don't see any sign of it in the code here, which is where it would need to happen.

**Pull request closed without merging**
